tests/phpunit/tests/rest-api/rest-posts-controller.php

```diff
@@ -5715,6 +5716,50 @@ public function test_get_posts_with_pagination() {
 		$this->assertErrorResponse( 'rest_post_invalid_page_number', $response, 400 );
 	}
 
+	/**
+	 * Test the REST API support for `ignore_sticky_posts`.
+	 *
+	 * @ticket 35907
+	 *
+	 * @covers WP_REST_Posts_Controller::get_items
+	 */
+	public function test_get_posts_ignore_sticky_default_prepends_sticky_posts() {
+		$id1 = self::$post_id;
+		// Create more recent post to avoid automatically placing other at the top.
+		$id2 = self::factory()->post->create( array( 'post_status' => 'publish' ) );
+
+		update_option( 'sticky_posts', array( $id1 ) );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertSame( $data[0]['id'], $id1, 'Response has sticky post at the top' );
+		$this->assertSame( $data[1]['id'], $id2, 'It is followed by most recent post' );
+	}
+
+	/**
+	 * Test the REST API support for `ignore_sticky_posts`.
+	 *
+	 * @ticket 35907
+	 *
+	 * @covers WP_REST_Posts_Controller::get_items
+	 */
+	public function test_get_posts_ignore_sticky_ignores_post_stickiness() {
+		$id1 = self::$post_id;
+		// Create more recent post to avoid automatically placing other at the top.
+		$id2 = self::factory()->post->create( array( 'post_status' => 'publish' ) );
+
+		update_option( 'sticky_posts', array( $id1 ) );
+
+		$request = new WP_REST_Request( 'GET', '/wp/v2/posts' );
+		$request->set_param( 'ignore_sticky', true );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertSame( $data[0]['id'], $id2, 'Response has no sticky post at the top' );
```

Comment:
If I'm reading the code correctly, `$data[1]` should be `$id1` so it would be good to test this.

Reply:
I had to double-check; it looks like the setup creates a couple more posts after `self::$post_id`, so it's not the next recent post.

Probably not relevant for this test. It just checks that the recently created post is the latest and the sticky post isn't prepended.